### ISSUE TITLE
When should one use 'hidden' in 'changelog-sections'?

### ISSUE THREAD

**Issue by davidhsingyuchen** [googleapis/release-please#1745]
## TL;DR

Instead of a `hidden` section in `changelog-sections`, why doesn't one just remove the section? Is there any practical difference?

## Background

I'm trying to customize the definition of releasable units, and I realized that the types defined in `changelog-sections` are essentially releasable units. As a result, I'm checking what's inside `changelog-sections`, and I found `hidden`. I did some experiments and realized that if a section is `hidden`, then the corresponding type is not a releasable unit anymore, hence the question in `TL;DR` above. I read [the description of `hidden` in the JSON schema](https://github.com/googleapis/release-please/blob/408ddac5dd4e1bc2c9b992365fa864e80298cee5/schemas/config.json#L41-L43), but it does not seem to mention _when_ `hidden` can be useful.

Thank you for reading, and please let me know if I'm missing anything, thanks!

**Comment by jthegedus** [comment]
I wanted to show my `docs:` commits in the Changelog, so documentation updates are listed alongside each release, so I set `{ "type": "docs", "section": "Documentation", "hidden": false }`.

But now `docs:` commits are being considered for in the calculation for version bumps, which is not the expected behaviour. I would expect these to be independent behaviours.
